ironfish/src/wallet/account.ts

```diff
@@ -213,6 +216,156 @@ export class Account {
     })
   }
 
+  private async saveConnectedMintsToAssetsStore(
+    transaction: Transaction,
+    tx: IDatabaseTransaction,
+  ): Promise<void> {
+    for (const { asset, value } of transaction.mints) {
+      // Only store the asset for the owner
+      if (asset.owner().toString('hex') !== this.publicAddress) {
+        continue
+      }
+
+      const assetId = asset.id()
+      const existingAsset = await this.walletDb.getAsset(this, assetId, tx)
+
+      let createdTransactionHash = transaction.hash()
+      let supply = BigInt(0)
+      if (existingAsset) {
+        createdTransactionHash = existingAsset.createdTransactionHash
+        supply = existingAsset.supply
+      }
+
+      await this.walletDb.putAsset(
+        this,
+        assetId,
+        {
+          createdTransactionHash,
+          id: assetId,
+          metadata: asset.metadata(),
+          name: asset.name(),
+          owner: asset.owner(),
+          supply: supply + value,
+        },
+        tx,
+      )
+    }
+  }
+
+  private async saveConnectedBurnsToAssetsStore(
+    transaction: Transaction,
+    tx: IDatabaseTransaction,
+  ): Promise<void> {
+    for (const { assetId, value } of transaction.burns) {
+      const existingAsset = await this.walletDb.getAsset(this, assetId, tx)
+      if (!existingAsset) {
+        continue
+      }
```

Comment:
This shouldn't happen, but what should happen if the wallet processes a burn transaction for an asset it owns, but the asset is missing from the db?

Reply:
That seems like an invalid db state, so I'd expect an assertion to fail in that case. Similar to how we look up transactions via a note and throw assertion errors. We should do the same here. 

This block skips in a loop because this method will be called for all accounts in a wallet. I wouldn't want a crash if Account A is burning an asset it owns but Account B sees the burn